JOING-57 feat: AI 통신간 변경사항 반영

hyoom opened the pull request:
## #️⃣ 연관된 이슈

> #JOING-57

<br/>

## 📝 작업 내용

> AI 통신간 변경사항 반영했습니다.   AI 통신간 4xx 상태 코드별 분기 처리, 필드명 일부 수정, 엔드포인트 일부 수정했습니다.

<br/>

## 📷 스크린샷(선택)

> 이미지가 있다면 첨부해주세요

<br/>

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?
>

hyoom committed:
JOING-57 feat: AI 통신간 4xx 상태 코드별 분기 처리 추가

hyoom committed:
JOING-57 fix: AI 통신간 필드명 일부 수정

hyoom committed:
JOING-57 feat: AI 통신간 엔드포인트 일부 수정

jayden-bin commented on src/main/java/com/ktb/joing/common/config/SecurityConfig.java:
```diff
@@ -66,14 +66,16 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .authorizeHttpRequests(auth -> auth
                         .requestMatchers("/", "/healthz", "/oauth2/**", "/login/**").permitAll()
-                        .requestMatchers("/api/v1/users/evaluation/**").permitAll()
+                        .requestMatchers("/api/v1/users/evaluation/**").permitAll() // AI 서버 관련 엔드 포인트
+                        .requestMatchers("/api/v1/items/*/evaluation").permitAll()
+                        .requestMatchers("/api/v1/items/*/summary").permitAll()
```
으흠 요 API들도 로그인 처리가 필요해 보이는데 permitAll을 사용하는 이유가 있을까요?

hyoom replied:
현재 Security 인증 전파에 문제가 있습니다. 이를 해결하는 데 시간이 조금 걸릴 것으로 예상되어 우선적으로 AI와 통신하는 부분에 대해 permitAll로 설정해 두었습니다... 이후 문제를 해결한 뒤, 해당 설정은 제거할 예정입니다!

jayden-bin commented on src/main/java/com/ktb/joing/common/util/webClient/ReactiveHttpService.java:
```diff
@@ -12,18 +13,38 @@
+                                    try {
+                                        AIErrorResponse errorResponse = objectMapper.readValue(errorBody, AIErrorResponse.class);
+                                        AiErrorCode errorCode = switch (errorResponse.getDetail()) {
+                                            case "유효하지 않은 형식의 채널아이디입니다." -> AiErrorCode.AI_INVALID_CHANNEL_ID_FORMAT;
+                                            case "유효하지 않은 채널아이디입니다." -> AiErrorCode.AI_INVALID_CHANNEL_ID;
+                                            case "영상의 개수가 충분하지 않아 더이상의 평가가 불가능합니다." -> AiErrorCode.AI_INSUFFICIENT_VIDEOS;
```
에러 분기를 메시지로 하는건 별로 좋은 선택지가 아니예요 ㅎㅎㅎ...
별도의 errorMessageCode를 정의해서 통신하는걸 권장해드립니다!

hyoom replied:
AI에서 오는 오류 응답이 메시지 형태로만 제공되어 이렇게 진행했었습니다... 말씀 주신 대로 AI 팀과 상의하여 별도의 Error Code를 정의하고 이를 기반으로 통신하도록 수정하겠습니다! 감사합니다 ~

hyoom committed:
JOING-57 fix: 요약 재생성 로직 변경

- Summary 업데이트 시 매번 새로운 객체를 생성하여 unique constraint violation 문제 발생
- Summary 엔티티에 부분 업데이트를 지원하는 update 메서드 추가
- 부분 업데이트가 불가능한 구조에서 기존 Summary가 있을 경우 새로 생성하지 않고 부분 업데이트하도록 수정

hyoom committed:
JOING-57 refactor: AI 에러 통신 처리 방식 개선

- 에러 메시지 기반에서 에러 코드 기반으로 개선
